cpu/stm32/periph/flashpage.c

```diff
@@ -234,6 +236,12 @@ void flashpage_write(void *target_addr, const void *data, size_t len)
     bool instruction_cache = FLASH->ACR & FLASH_ACR_ICEN;
     if (instruction_cache) {
         FLASH->ACR &= ~FLASH_ACR_ICEN;
+#if defined(CPU_FAM_STM32WL)
+        /* Reset the instruction cache after it has been disabed. This is
+           required for the proper working of flash I/O operations while LPUART
+           is active */
```

Comment:
This is unrelated to LPUART and to UART, without this the CPU stalls, so the comment is still a bit misleading. IMO it should mention that without this `stm32wl` stalls after a flash write operation for unknown reasons.